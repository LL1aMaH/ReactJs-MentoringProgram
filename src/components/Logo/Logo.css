a {
  color: var(--primary-link-color);
  font-size: 25px;
  text-decoration: none;

  & b {
    font-weight: 900;
  }
}
