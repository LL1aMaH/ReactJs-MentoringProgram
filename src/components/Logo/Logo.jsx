import React, { memo } from 'react';

import classnames from 'classnames';

import styles from '../../../styles/Logo.module.scss' 

export const Logo = memo(function Logo({ className }) {
  return (
    <a href="/" className={classnames(styles.logo, className)} id="a">
      <b>netflix</b>roulette
    </a>
  );
});
