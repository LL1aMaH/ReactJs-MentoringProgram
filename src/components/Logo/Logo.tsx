import React, { memo } from 'react';

import { Link } from 'react-router-dom';
import classnames from 'classnames';

import styles from './Logo.css';

export type LogoProp = {
  className?: string;
};

export const Logo = memo(function Logo({ className }: LogoProp): JSX.Element {
  return (
    <Link to="/" className={classnames(styles.logo, className)}>
      <b>netflix</b>roulette
    </Link>
  );
});
