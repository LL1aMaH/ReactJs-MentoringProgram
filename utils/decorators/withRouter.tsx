import React from 'react';

import { MemoryRouter } from 'react-router-dom';

export type TComponentFunc = () => JSX.Element;
export type TDecoratorLocation = {
  pathname?: string;
  search?: string;
};

const withRouter = (
  componentFunc: TComponentFunc,
  location: TDecoratorLocation = { pathname: '/' },
): JSX.Element => {
  return <MemoryRouter initialEntries={[location]}>{componentFunc()}</MemoryRouter>;
};

export default withRouter;
