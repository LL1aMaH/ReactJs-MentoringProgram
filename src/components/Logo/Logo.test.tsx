import React from 'react';
import { render } from '@testing-library/react';

import withRouter from '../../../utils/decorators/withRouter';

import { Logo } from './Logo';

describe('NotFoundPage', () => {
  it('renders correctly', () => {
    const component = render(withRouter(() => <Logo />));
    expect(component.asFragment()).toMatchSnapshot();
  });
});
