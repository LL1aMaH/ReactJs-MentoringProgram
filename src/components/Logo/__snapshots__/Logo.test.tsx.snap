// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundPage renders correctly 1`] = `
<DocumentFragment>
  <a
    class="logo"
    href="/"
  >
    <b>
      netflix
    </b>
    roulette
  </a>
</DocumentFragment>
`;
